### PR TITLE
chore: make clean should clean tools/bin

### PR DESCRIPTION
**Pull request by zirain**

### CODE REVIEW

**Review by arkodg — COMMENTED** [review]
lets add a `tools.clean` and add this `rm` under it, with the definition living under 
https://github.com/envoyproxy/gateway/blob/main/tools/make/tools.mk

File: tools/make/tools.mk [hunk 1]
```diff
@@ -63,3 +63,11 @@ tools/bin/$(notdir $(SHELLCHECK_TXZ)):
 	mkdir -p $(@D)
 	tar -C $(@D) -Jxmf $< --strip-components=1 shellcheck-v$(SHELLCHECK_VERSION)/shellcheck
 endif
+
+tools.clean: # Remove all tools
+	@$(LOG_TARGET)
+	rm -rf $(tools.bindir)
+
+.PHONY: clean
+clean: ## Remove all files that are created during builds.
```
Comment (arkodg) [inline tools/make/tools.mk:72]: umm there's already a `clean` target under `go.mk`

Comment (zirain) [reply tools/make/tools.mk:72]: there's another one in `docs.mk`, just follow the old way.

Comment (arkodg) [reply tools/make/tools.mk:72]: ah so `make clean` will call all of them ?

**Review by arkodg — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]

**Pull request merged**